### PR TITLE
chore: Fix running config::enterprise tests

### PR DESCRIPTION
**Pull request by wbew**
This PR
- Allows the enterprise tests to run in CI properly by re-arranging the compilation feature gates. 

For context, I'd realized, while working on #13355 that these weren't running since they were configured under the `enterprise-tests` feature which does not activate when tests are run in CI. Along the way, I've split the enterprise tests up into unit tests and behavioral/integration tests and wired the latter to run in `test-enterprise`.

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Nice catch. Just a couple of thoughts.

File: Makefile [hunk 1]
```diff
@@ -318,13 +318,17 @@ test-behavior-config: ## Runs configuration related behaviorial tests
...
+	${MAYBE_ENVIRONMENT_EXEC} cargo test --features "default, enterprise-tests" -- config::enterprise
+
 .PHONY: test-behavior-%
 test-behavior-%: ## Runs behaviorial test for a given category
 	${MAYBE_ENVIRONMENT_EXEC} cargo run -- test tests/behavior/$*/*
 
 .PHONY: test-behavior
 test-behavior: ## Runs all behaviorial tests
-test-behavior: test-behavior-transforms test-behavior-formats test-behavior-config
+test-behavior: test-behavior-transforms test-behavior-formats test-behavior-config test-behavior-enterprise
```
[the remark this answers is not in the document]
Comment (001wwang) [reply Makefile:331]: Ah, great point, will relocate and revamp this.

File: Makefile [hunk 2]
```diff
@@ -318,13 +318,17 @@ test-behavior-config: ## Runs configuration related behaviorial tests
 	${MAYBE_ENVIRONMENT_EXEC} cargo build --bin secret-backend-example
 	${MAYBE_ENVIRONMENT_EXEC} cargo run -- test tests/behavior/config/*
 
+.PHONY: test-behavior-enterprise
+test-behavior-enterprise: ## Runs enterprise related behaviorial tests
+	${MAYBE_ENVIRONMENT_EXEC} cargo test --features "default, enterprise-tests" -- config::enterprise
```
[the remark this answers is not in the document]
Comment (001wwang) [reply Makefile:323]: Thanks for catching that, I was experimenting with something and forgot to remove it. Will fix.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (001wwang) [reply Makefile]: Created [`make test-enterprise`](https://github.com/vectordotdev/vector/pull/13369/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R330) and wired it up in a few places. I find these tests are on the fine line of not quite unit tests but also not quite integration tests (i.e. no external services), so not sure what the best setup is for them. Let me know what you think, happy to change it.

**Review by jszwedko — APPROVED** [review]
Looks good!

**Pull request merged**